test/plugin/test_out_google_cloud.rb

```diff
@@ -856,310 +70,36 @@ def test_server_error
-
-  def test_struct_container_log_metadata_from_tag
-    setup_gce_metadata_stubs
-    setup_container_metadata_stubs
-    setup_logging_stubs
-    d = create_driver(APPLICATION_DEFAULT_CONFIG, CONTAINER_TAG)
-    d.emit(container_log_entry('{"msg": "test log entry 0", ' \
-                               '"tag2": "test", "data": 5000, ' \
-                               '"severity": "W"}'))
-    d.run
-    verify_log_entries(1, CONTAINER_FROM_TAG_PARAMS,
-                       'structPayload') do |entry|
-      assert_equal 3, entry['structPayload'].size, entry
-      assert_equal 'test log entry 0', entry['structPayload']['msg'], entry
-      assert_equal 'test', entry['structPayload']['tag2'], entry
-      assert_equal 5000, entry['structPayload']['data'], entry
-      assert_equal CONTAINER_SECONDS_EPOCH, \
-                   entry['metadata']['timestamp']['seconds'], entry
-      assert_equal CONTAINER_NANOS, \
-                   entry['metadata']['timestamp']['nanos'], entry
-      assert_equal 'WARNING', entry['metadata']['severity'], entry
-    end
-  end
-
-  def test_one_cloudfunctions_log
-    setup_gce_metadata_stubs
-    setup_cloudfunctions_metadata_stubs
-    setup_logging_stubs
-    d = create_driver(APPLICATION_DEFAULT_CONFIG, CLOUDFUNCTIONS_TAG)
-    d.emit(cloudfunctions_log_entry(0))
-    d.run
-    verify_log_entries(1, CLOUDFUNCTIONS_PARAMS) do |entry|
-      assert_equal 'DEBUG', entry['metadata']['severity'], entry
-    end
-  end
-
-  def test_multiple_cloudfunctions_logs
-    setup_gce_metadata_stubs
-    setup_cloudfunctions_metadata_stubs
-    setup_logging_stubs
-    d = create_driver(APPLICATION_DEFAULT_CONFIG, CLOUDFUNCTIONS_TAG)
-    [2, 3, 5, 11, 50].each do |n|
-      # The test driver doesn't clear its buffer of entries after running, so
-      # do it manually here.
-      d.instance_variable_get('@entries').clear
-      @logs_sent = []
-      n.times { |i| d.emit(cloudfunctions_log_entry(i)) }
-      d.run
-      verify_log_entries(n, CLOUDFUNCTIONS_PARAMS) do |entry|
-        assert_equal 'DEBUG', entry['metadata']['severity'], entry
-      end
-    end
-  end
-
-  def test_one_cloudfunctions_log_text_not_matched
-    setup_gce_metadata_stubs
-    setup_cloudfunctions_metadata_stubs
-    setup_logging_stubs
-    d = create_driver(APPLICATION_DEFAULT_CONFIG, CLOUDFUNCTIONS_TAG)
-    d.emit(cloudfunctions_log_entry_text_not_matched(0))
-    d.run
-    verify_log_entries(1, CLOUDFUNCTIONS_TEXT_NOT_MATCHED_PARAMS) do |entry|
-      assert_equal 'INFO', entry['metadata']['severity'], entry
-    end
-  end
-
-  def test_multiple_cloudfunctions_logs_text_not_matched
+  # This test looks similar between the grpc and non-grpc paths except that when
+  # parsing "105", the grpc path responds with "DEBUG", while the non-grpc path
+  # responds with "100".
+  #
+  # TODO(lingshi) - consolidate the tests between the grpc path and the non-grpc
+  # path: split it into two tests, one with string severities and one with
```

Comment:
No need for the "-" after the TODO, and splitting is a kind of consolidation (because the string severity path is common). How about:
```
# TODO(lingshi) consolidate the tests between the grpc path and the non-grpc
# path, or at least split into two tests, one with string severities and one with
# numeric severities.
```
I'd also replicate the TODO into the other file.